ForgeUnity/Assets/BeardedManStudios/Scripts/StandAlone/Objects/InterpolateFloat4.cs

```diff
@@ -0,0 +1,21 @@
+﻿using BeardedManStudios.Forge.Networking;
+
+namespace BeardedManStudios
+{
+    public struct InterpolateFloat4 : IInterpolator<Float4>
+    {
+        public Float4 current;
+        public Float4 target;
+        public float LerpT { get; set; }
+        public bool Enabled { get; set; }
+        public ulong Timestep { get; set; }
+
+        public Float4 Interpolate()
+        {
+            if (!Enabled) return target;
+
+            current = Float4.Lerp(current, target, LerpT);
```

Comment:
I've discovered that this is problematic if you're using a `Float4` to replace a Quaternion, as it uses `Slerp` instead. There will either need to be multiple choices for interpolation, or (easier) a separate non-Unity type for a quaternion.

Switching to .NET Numeric's classes may be opportune for the latter, then we can use Vector2, Vector3, Vector4 and Quaternion to avoid introducing yet another custom type.

Reply:
Fixed by switching to `System.Numeric.Quaternion`, see below.